tiatoolbox/dataloader/wsireader.py

```diff
@@ -18,28 +18,37 @@
 # All rights reserved.
 # ***** END GPL LICENSE BLOCK *****
 
-"""WSIReader for WSI reading or extracting metadata information from WSIs."""
+"""This module defines classes which can read image data from WSI formats."""
 from tiatoolbox.utils import misc, transforms
 from tiatoolbox.dataloader.wsimeta import WSIMeta
 
 import pathlib
+import warnings
 import numpy as np
 import openslide
 import glymur
 import math
 import pandas as pd
+import cv2
 import re
-import warnings
+import numbers
+import os
+
+glymur.set_option("lib.num_threads", os.cpu_count() or 1)
 
 
 class WSIReader:
-    """WSI Reader class to read WSI images.
+    """Base whole slide image (WSI) reader class.
+
+    This class defines functions for reading pixel data and metadata
+    from whole slide image (WSI) files.
 
     Attributes:
-        input_path (pathlib.Path): input path to WSI directory.
-        output_dir (pathlib.Path): output directory to save the output.
-        tile_objective_value (int): objective value at which tile is generated.
-        tile_read_size (int): [tile width, tile height]
+        input_path (pathlib.Path): Input path to WSI file.
+        output_dir (pathlib.Path): Output directory to save the output.
+        tile_objective_value (int): Objective value at which tile is generated.
+        tile_read_size_w (int): Tile width.
```

Comment:
Shall we use a tuple here?

Reply:
For tile with and height? I didn't change this. Was it from the merge? I propose we remove `output_dir`, `tile_objective_value`, `tile_read_size`, and `output_dir` from the init and have them as arguments to some tile function elsewhere. They do not need to be known at init time. Best to avoid asking for args which are not required at this point.

Reply:
It might have been from merge as I prioritised docstring from develop in merge but may have missed to notice the change.